docs: author changelog (2nd iteration)

0xFA11 opened the pull request:
Following up on previous iteration (#1087), this PR adds other missing entries to `CHANGELOG.md`.

JIRA Ticket: MTT-1239

## Changelog

### com.unity.netcode.gameobjects

- _(This PR updates the `CHANGELOG.md` with missing entries.)_

## Testing and Documentation

- N/A

0xFA11 committed:
add new changelog todo entries

0xFA11 committed:
minor reorg update

0xFA11 committed:
fatih's pass

0xFA11 committed:
matt's pass

SamuelBellomo commented on com.unity.netcode.gameobjects/CHANGELOG.md:
```diff
@@ -10,6 +10,8 @@ Additional documentation and release notes are available at [Multiplayer Documen
 ### Added
 
+- Added `ClientNetworkTransform` sample to the SDK package (#1168)
+- Added `Bootstamp` sample to the SDK package (#1140)
```
```suggestion
- Added `Bootstrap` sample to the SDK package (#1140)
```

SamuelBellomo replied:
lol

SamuelBellomo commented on com.unity.netcode.gameobjects/CHANGELOG.md:
```diff
@@ -93,6 +108,8 @@ Jaedyn:
+- [d8a3a9b3] (2021-09-20) Sam Bellomo / feat: ClientNetworkTransform (#1177)
 - [5513c906] (2021-09-16) Sam Bellomo / feat: interpolation for network transform (#1060)
 - [cc7a7d5c] (2021-08-19) Sam Bellomo / test: adding more details to multiprocess readme (#1050)
 - [13e2b7f1] (2021-07-13) Sam Bellomo / test: multiprocess tests part 6: fixing issues runnings all tests together (#957)
```
if changes are not user facing, do we still need to document them here?

0xFA11 replied:
no. the goal is to document user facing and effecting changes. any behavioral changes users need to know about or any public API changes etc should be documented here. internal, not user facing/effecting changes, chores etc does not have to be documented in the changelog. you might take a look at other entries in the list :)

SamuelBellomo replied:
ok, then we'd need to review this list, all the multiprocess tests stuff isn't userfacing.
There's also a few items that are more internal iteration that users won't care about (they'll only care about the end product, not how we got there)

0xFA11 committed:
minor grammar fix

0xFA11 committed:
Merge branch 'develop' of github.com:Unity-Technologies/com.unity.netcode.gameobjects into docs/changelog-2

0xFA11 committed:
update & remove some items from the TODO list

NoelStephensUnity committed:
Updated Noel Stephens

Updated the entries that were pertinent to users and removed the ones that were just working out bugs from new V1.0.0 features.

0xFA11 committed:
add new todo items

0xFA11 committed:
Merge branch 'develop' into docs/changelog-2

ShadauxCat committed:
Changelog for Jaedyn

jeffreyrainy committed:
changelog entries for Jeff

SamuelBellomo committed:
Changelog for Sam

SamuelBellomo commented on com.unity.netcode.gameobjects/CHANGELOG.md:
```diff
@@ -128,80 +137,29 @@ Philipp:
-- [3d5b29a0] (2021-09-17) Andrew Spiering / chore: updating UTP package to pull in DTLS fixes (#1197)
+- [058fdb55] (2021-10-11) Luke Stampfli / feat: Client networkrigidbody (#1202)
+- [0b80225d] (2021-10-08) Jesse Olmer / chore: transport adapter to 0.1.0-preview.1 (#1274)
+- [13478086] (2021-10-06) LPLafontaineB / Changed the transport for the Bootstrap sample, from Unet to UTP (#1268)
```
```suggestion
```

SamuelBellomo committed:
removing LP's entry, it's linked to bootstrap sample already in this changelog

0xFA11 committed:
merge branch 'develop' into 'docs/changelog-2'

0xFA11 committed:
minor updates

LukeStampfli committed:
luke changelog entries

0xFA11 committed:
deleted more items from the todo list

0xFA11 committed:
merge branch 'develop' into 'docs/changelog-2'

0xFA11 committed:
albin's entries

0xFA11 committed:
merge branch 'develop' into 'docs/changelog-2'

Commit:
Merge branch 'develop' into docs/changelog-2
